addons/mail/models/discuss/discuss_channel_category.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class ChannelCategory(models.Model):
+    _name = "discuss.channel.category"
+    _description = "Channel Category"
+
+    name = fields.Char('Channel category name', required=True)
+    channel_ids = fields.One2many('discuss.channel', 'channel_category_id', string='Channels')
+
+    @api.model
+    def get_categories(self):
+        categories = self.env['discuss.channel.category'].search([])
```

Comment:
This is also prone to scaling issue if there are many categories.
I don't really see how well we could return only some of them.
Maybe just returning the category_id on channels would be enough?